basyx/aas/model/base.py

```diff
@@ -2272,13 +2270,9 @@ def _set_unit(self, unit: Optional[str]):
         """
         Check the input string
 
-        Constraint AASd-100: An attribute with data type "string" is not allowed to be empty
-
         :param unit: unit of the data object (optional)
         :raises ValueError: if the constraint is not fulfilled
         """
-        if unit == "":
-            raise ValueError("unit is not allowed to be an empty string")
```

Comment:
I think we should keep these `ValueErrors` as we otherwise have an ambiguity with `None` and an empty string.

Reply:
This issue will be solved automatically, when implementing AASd-130 (#118) :

> Constraint AASd-130: An attribute with data type "string" shall consist of these
characters only: ^[\x09\x0A\x0D\x20-\uD7FF\uE000-\uFFFD\u00010000-
\u0010FFFF]*$.

Reply:
How? As far as I can tell this regex allows the empty string.

Reply:
Right, but in combination with the constrained string type regex, it should not be allowed. (Otherwise that's another finding)

Reply:
Yes, but `unit`, `sourceOfDefinition` and `symbol` are not of any constrained string type, but simply of type `string`.
But I also just noticed that while Constraint AASd-100 seens to have been removed from the DataSpecification classes, it is still part of Part1, where it doesn't make any sense anymore, since Part1 doesn't contain any attributes of type `string`. But assuming it still applies to Part3a, should this be left as is or simply changed to an `AASConstraintViolation(100, ...)` instead of a `ValueError()`?